src/pages/Transfer/CrossChainTransferForm/index.tsx

```diff
@@ -1,52 +1,187 @@
+
+    try {
+      setSubmitStatus(STATUSES.PENDING);
+
+      await xcmTransfer(
+        address,
+        destination.address,
+        newMonetaryAmount(data[TRANSFER_AMOUNT], COLLATERAL_TOKEN, true)
+      );
+
+      setSubmitStatus(STATUSES.RESOLVED);
+    } catch (error) {
+      setSubmitStatus(STATUSES.REJECTED);
+      setSubmitError(error);
+    }
+  };
+
+  const handleConfirmClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    if (!address) {
+      dispatch(showAccountModalAction(true));
+      event.preventDefault();
+    }
+  };
+
+  // This ensures that triggering the notification and clearing
+  // the form happen at the same time.
+  React.useEffect(() => {
+    if (submitStatus !== STATUSES.RESOLVED) return;
+
+    toast.success(t('transfer_page.successfully_transferred'));
+
+    reset({
+      [TRANSFER_AMOUNT]: ''
+    });
+  }, [
+    submitStatus,
+    reset,
+    t
+  ]);
+
+  const handleValueChange = (event: any) => {
+    const value = newMonetaryAmount(event.target.value, COLLATERAL_TOKEN, true);
+    const usd = getUsdAmount(value, prices.governanceToken.usd);
+
+    setApproxUsdValue(usd);
+  };
+
   return (
-    <form className='space-y-8'>
-      <FormTitle>
-        {t('transfer_page.cross_chain_transfer_form.title')}
-      </FormTitle>
-      <div>
-        <AvailableBalanceUI
-          label={t('transfer_page.cross_chain_transfer_form.balance')}
-          balance={displayMonetaryAmount(collateralTokenTransferableBalance)} />
+    <>
+      <form
+        className='space-y-8'
+        onSubmit={handleSubmit(onSubmit)}>
+        <FormTitle>
+          {t('transfer_page.cross_chain_transfer_form.title')}
+        </FormTitle>
+        <div>
+          <div>
+            <TokenField
+              onChange={handleValueChange}
+              id={TRANSFER_AMOUNT}
+              name={TRANSFER_AMOUNT}
+              ref={register({
+                required: {
+                  value: true,
+                  message: t('transfer_page.cross_chain_transfer_form.please_enter_amount')
+                }
+              })}
+              error={!!errors[TRANSFER_AMOUNT]}
+              helperText={errors[TRANSFER_AMOUNT]?.message}
+              label={COLLATERAL_TOKEN_SYMBOL}
+              approxUSD={`≈ $ ${approxUsdValue}`} />
+          </div>
+        </div>
+        <div>
+          <Chains
+            label={t('transfer_page.cross_chain_transfer_form.from_chain')}
+            defaultChain={ChainType.RelayChain} />
+        </div>
+        <div>
+          <Chains
+            label={t('transfer_page.cross_chain_transfer_form.to_chain')}
+            defaultChain={ChainType.Parachain} />
+        </div>
         <div>
-          <TokenField
-            label={COLLATERAL_TOKEN_SYMBOL}
-            approxUSD='≈ $ 0' />
+          <Accounts
+            label={t('transfer_page.cross_chain_transfer_form.target_account')}
+            callbackFunction={setDestination} />
```

Comment:
There are bigger problems than this with react hook form - I spent some time looking at this on a previous task, and because the select component doesn't ultimately render a select, passing the ref down using react-hook-form was not great from a code point of view. Ultimately we need to refactor how we use select fields in forms so that they can be used with react-hook-form or similar, but for the purposes of this feature I'm happy with the approach we have here.

Reply:
@tomjeatt 
For now, it's fine in consideration of tight timeline. But I'd say we should refactor it properly. Other than ref passing issue, it has also redundant internal state issue as well.
If you want, I can refactor it properly later. Let me create a ticket.